src/arch/aarch64/kernel/processor.rs

```diff
@@ -221,27 +207,14 @@ pub fn detect_frequency() {
 fn __set_oneshot_timer(wakeup_time: Option<u64>) {
 	if let Some(wt) = wakeup_time {
 		// wt is the absolute wakeup time in microseconds based on processor::get_timer_ticks.
-		let deadline = u128::from(wt) * u128::from(CPU_FREQUENCY.get()) / 1_000_000;
-		let deadline = u64::try_from(deadline).unwrap();
+		let deadline = (wt * u64::from(get_frequency())) / 1000;
```

Comment:
I assume this is fine, but I'd suggest double-checking that a `u64` is sufficient here (was `u128` before).

Reply:
move this part to #1703